### PR TITLE
Use bisect instead of iterating over every offsets.

### PR DESCRIPTION
**Pull request by JulienPalard**
As the offsets looks already sorted (which looks logical so I assumed
they always are), using a bisection if faster than iterating over all
of them.

On a "worst case scenario" I encontered, I got nice enhancement with this patch:

$ time python3 ./pycodestyle.py ~/Downloads/PmagPy/PmagPy/coefficients-552K.py > /dev/null

real 1m16.405s

$ time python3 ./pycodestyle.py ~/Downloads/PmagPy/PmagPy/coefficients-552K.py > /dev/null

real 0m3.318s

### CODE REVIEW

File: pycodestyle.py [hunk 1]
```diff
@@ -1645,10 +1646,10 @@ def check_logical(self):
         """Build a line from tokens and run all logical checks on it."""
         self.report.increment_logical_line()
         mapping = self.build_tokens_line()
-
         if not mapping:
             return
 
+        mapping_offsets = [offset for offset, pos in mapping]
```
Comment (sigmavirus24) [inline pycodestyle.py]: There's no need to bind `pos` here. This should more simply be

```py
mapping_offsets = [offset for offset, _ in mapping]
```

File: pycodestyle.py [hunk 2]
```diff
@@ -1662,9 +1663,8 @@ def check_logical(self):
             self.init_checker_state(name, argument_names)
             for offset, text in self.run_check(check, argument_names) or ():
                 if not isinstance(offset, tuple):
-                    for token_offset, pos in mapping:
-                        if offset <= token_offset:
-                            break
+                    token_offset, pos = mapping[bisect.bisect_left(
```
Comment (sigmavirus24) [inline pycodestyle.py]: I'd prefer it if you added a comment here explaining the benefit of `bisect_left` so this doesn't get replaced later on by a well-meaning person.

File: pycodestyle.py (hunk 1 above)
Comment (JulienPalard) [inline pycodestyle.py]: fair enough

File: pycodestyle.py (hunk 2 above)
Comment (JulienPalard) [inline pycodestyle.py]: done.

**Comment by JulienPalard** [comment]
Fixed your thoughts and rebased on master.

**Comment by sigmavirus24** [comment]
FWIW, thank you so much for including everything in the commit message. It means that if this project survives long enough for GitHub to go the way of Sourceforge that we'll still have an understanding of why we did this. 🎉 🍰

**Pull request merged**